src/components/oauthClient.ts

```diff
@@ -54,17 +75,29 @@ async function postWithAuth<ExpectedDataType>(
 ): Promise<{ status: number; data: ExpectedDataType }> {
   const authHeaders = await getAuthHeader();
   return await fetch(url, {
-    method: 'POST',
-    headers: { ...authHeaders, Accept: 'application/json', 'Content-Type': 'application/json' },
+    method: "POST",
+    headers: {
+      ...authHeaders,
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
     body: body ? JSON.stringify(body) : undefined,
   }).then(async (res) => {
+    if (!res.ok) {
+      console.error(`[FetchError] Request failed`, {
+        url,
+        status: res.status,
+        statusText: res.statusText,
+      });
+      return { status: res.status, data: {} };
+    }
     return { status: res.status, data: await res.json() };
```

Comment:
in this case it is worth observing the "no negative conditions" directive
(edit: no need for back-ticks here)
```suggestion
    if (res.ok) {
      return { status: res.status, data: await res.json() };
    }

console.error('[FetchError] Request failed', {
  url,
  status: res.status,
  statusText: res.statusText,
});

return { status: res.status, data: {} };
```

Reply:
good call. code changed.

Reply:
perhaps this and the counter starting at 0 are uncommitted changes? I can't see either